test/test_iterdatapipe.py

```diff
@@ -889,6 +889,132 @@ def mul_fn(a, b):
         with self.assertRaisesRegex(TypeError, "length relies on the output of its function."):
             len(flatmapped_dp)
 
+    def test_shuffled_flatmap_iterdatapipe(self):
+        source_dp = IterableWrapper(list(range(20)))
+
+        def fn(e):
+            return [e, e * 10]
```

Comment:
Can we add one more test with generator function? I believe that's the reason we still keep it as `iter`.

Reply:
Done 👌